workspaces/analytics/src/events/diffs.ts

```diff
@@ -1,6 +1,6 @@
 import { DescribeEvent } from '../interfaces/RegisterEvent';
```

Comment:
I started touching this file but realized I didn't need to be here. That being said, I'm not a huge fan of the JOI structure for a couple of reasons:
- We aren't using the typing correctly. The idea behind this seems to be to ensure consistent typing / event structures - we can achieve the same thing with typescript with a lot less indirection (since typescript will statically check)
- Our UI events are using a completely different thing

What I'm planning to do in a future PR (unless there's any pushback here) is to:
- Remove context from the `TrackEventsBase` (and pass in context automatically)
- Consolidate UI events + CLI events - these are effectively just - an event name and an object, we don't need two different ways to declaring events.
- Build a lighter weight layer with better typesafety - we can probably remove a bunch of this stuff by just specifying event creators (functions with arguments, or even just regular object types and typescript can statically check these)

The most important thing here is making sure our events are backwards compatible (for the ones we care about)